Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -1492,101 +1493,202 @@ google_firestore_v1_MapValue Serializer::EncodeMapValue(
   return result;
 }
 
-// TODO(varconst): everything below
-
 MutationResult Serializer::DecodeMutationResult(
+    nanopb::Reader* reader,
     const google_firestore_v1_WriteResult& write_result,
     const SnapshotVersion& commit_version) const {
-  // TODO(varconst): implement.
-  // - (MutationResult)decodedMutationResult:(GCFSWriteResult *)mutation
-  //     commitVersion:(const SnapshotVersion &)commitVersion;
-  return MutationResult{{}, {}};
+  // NOTE: Deletes don't have an update_time, use commit_version instead.
+  SnapshotVersion version =
+      write_result.has_update_time
+          ? DecodeSnapshotVersion(reader, write_result.update_time)
+          : commit_version;
+  absl::optional<std::vector<FieldValue>> transform_results;
```

Comment:
Nit: can you add an empty line before this line? (I know it was a wall of text in the original)